Some improvements. Fixed the viewmodel snapping around when turning quickly.

KM-CM opened the pull request:
Why did I close and now am reopening the pull request? Because GitHub did some BS which made it self-close. I didn't close it. It just... did that. Somewhy. I really don't know. So now opening another one...

Now, about your points as stated [here.](https://github.com/buu342/GMod-BuuBaseRedone/pull/4)

1. Sorry. Just sorry, I've got nothin' else to say. EDIT: I do now. I had no idea it was going to your EMAIL. And I did that just in case (don't know why myself, just me being stupid). I had no intention to spam your email or make you hurry ("look through my changes faster").
2. I think tabs are better than four spaces, but well, fixed it to four spaces now.
3. You were not actually supposed to see the commits. Not as in they're secret, but as in I've did EXTREMELY small updates, and GitHub saying "KM_CM added a commit" (or whatever it's saying I forgot the exact wording) was never desired by me, it's just the GitHub GUI's thing.

KM-CM committed:
Add files via upload

Fixed indentation to four spaces instead of tabs

buu342 commented:
No worries. I'll take a look at the changes in this PR instead :slightly_smiling_face:

KM-CM commented:
Sorry for spamming if these messages going to your email, but still gonna say - there's nothing different, other than there being four spaces instead of tabs.

buu342 commented on README.md:
Sure thing! You can remove the comment in the parenthesis so that it doesn't end up in the final merge

buu342 commented on lua/autorun/buu_base_settings.lua:
Remove all of these changes since the last commit I made to master should make the spacing consistent.

buu342 commented on lua/effects/buu_muzzle.lua:
Remove all of these changes since the last commit I made to master should make the spacing consistent.

KM-CM replied:
Way ahead of you with using Notepad++ to replace ``\t`` with ``    ``!

buu342 commented on lua/weapons/weapon_buu_base2.lua:
I think in general, the changes you made are fine. However, I don't fully agree with what was done to the jumping animation. The full jump animation should only happen when you jump, as it doesn't make sense for the gun to do the initial curve when falling. I do think that the falling animation should use the speed you picked, however. Here's a video illustration of what I ended up tweaking on my end:

https://github.com/user-attachments/assets/b4f62b78-f90e-4504-ab88-c25ceb1f4965

I also personally think that `SWEP.flIronSightSwayMultiplierTarget` should have a non-zero initial value, maybe `0.5`?

KM-CM replied:
`flIronSightSwayMultiplierTarget` I exaggerated to just show you roughly what needed to be done. Try ironsighting and then turning left/right. You'll see untextured viewmodel parts. **It was never meant to be used with fully modeled weapons, but rather to fix that for the not fully modeled ones.**

Now, about the jumping animation... transitions, transitions. What I mean is, I consider the curve a transition. You don't like it? Merge my changes and change it - **I am not against tweaks, this is just how I feel like it should have been done.** EDIT: Now I remembered, when I was looking at the code and taking some for my base ( dude, you're the only adequate base owner IMO ), I thought it was snapping because of the lack of the curve, not because of the animation speed. So that's why I implemented that. I feel like you shouldn't remove it, but if you do, it's not really important. Both are nice, you have to pick one. My addon vibes with always curve, you might vibe with jump only curve.

EDIT 2: ELECTRIC BOOGALOO: Don't change `flIronSightSwayMultiplierTarget` as much, it should be very close to zero, for the reasons of showing untextured parts, as I have mentioned above. But for Far Cry 3 weapons which I see you love to make, you might wanna set it to something like `.5`, just be careful - despite not being untextured, there are still ironsights that will not point at wherever you're aiming properly, and some potential for the gun getting inside of you (though very unlikely!), so maybe something like `.1` for default and `.2` for FC3 guns? I am unsure because I have not playtested said numbers, just suggesting.

KM-CM committed:
Updates, and finally did the small fix on the bobbing stuff

Since my changes aren't gonna get merged that quickly, releasing this! :D

KM-CM committed:
Really small fix...

...of things that made me question Buu's sanity

KM-CM committed:
Half-Life 2 weapon remakes

-for people to have fun with.

KM-CM committed:
Removed self-credit just in case

buu342 reviewed (changes requested):
I would rather weapons not be bundled with the weapon base.

KM-CM committed:
Spacing...

...was inconsistent because Notepad++ is hellbent on tabs and Buu loves four spaces.

KM-CM commented:
> I would rather weapons not be bundled with the weapon base.

I would recommend you include them. If you don't, that's fine. But regardless, here are some reasons to do include them.

1. They are remakes of Half-Life 2 weapons. That is literally the first thing you should do with a Garry's Mod weapon base IMO.
2. They don't take that much disk space, they're just a couple **extremely short** text files.
3. They showcase the awesome capabilities of your base.

P.S. I am gonna be resolving the spacing issues without any commentary from now on. Because they're just spacing issues. Notepad++, "Find in Files", "\garrysmod\addons\GMod-BuuBaseRedone-Improved\lua\", "Search Mode" "Regular expression", "Find what :" "\t", "Replace with :", "`    `"

KM-CM committed:
Update weapon_buu_base2.lua

The crosshair spread is actually static. So that makes the crosshair inaccurate when moving. Also fixed the crosshair to display correct spread. There are probably left variables I didn't remove!

KM-CM committed:
Okay, Buu, I did what you wanted.

Now the curve only happens when you jump. That, and some other optimizations.

KM-CM committed:
As per Buu's request, Half-Life 2 weapons are not bundled with the base anymore

KM-CM committed:
As per Buu's request, Half-Life 2 weapons are not bundled with the base anymore

KM-CM committed:
As per Buu's request, Half-Life 2 weapons are not bundled with the base anymore

KM-CM committed:
As per Buu's request, Half-Life 2 weapons are not bundled with the base anymore

KM-CM committed:
As per Buu's request, Half-Life 2 weapons are not bundled with the base anymore

KM-CM commented:
I've did everything now. Weapons aren't bundled with the base, jumping animation only plays when actually jumping (I still think this is stupid), and... that's pretty much it.

KM-CM committed:
Holy crap that was a Lua error!

Now it's fixed.

KM-CM commented:
Welp, fixed that now! :D

KM-CM commented:
Now I'm truly done!

KM-CM commented:
Sorry for spamming (if this counts as it), and again, I'm not trying to get you to hurry, but... are you still alive? /j

KM-CM commented:
Now I know that you're alive since you starred *that one repository* lol

buu342 commented:
Like I said, this PR makes a lot of changes, some which I agree and some which I don't. It will take me an entire afternoon to properly review the code and bisect changes that I want. Frankly, I haven't had the time nor patience to do that, and I do not know when I will. I haven't touched Garry's Mod since October.

KM-CM commented:
Not a problem, not forcing you to hurry. It's just that you disappeared from everywhere for a lot of time lol

KM-CM committed:
Just now realizing the original crosshair code was lit

KM-CM committed:
Increased stability systems

I feel like I'm writing some real freakin' military grade system here xD

KM-CM commented:
I have a couple things I would like to note. They are not about your base, but rather if you want to mimic Far Cry 3.

- Far Cry 3 crosshair growth is different from yours.
- It also fades out after growing (do note that my addon, KM_CM's Addon, has the almost exact feeling fading out of the crosshair, but you'll have to dig in there yourself, as I am not trying to self-promote 'ere).
- Nearwall doesn't exist. Far Cry 3 has a cover system (my addon has it too- *gets shot 67 times*), which is why certain guns go in such positions.

And specifically about your base, for weapons which aren't "fully modeled" as I call them (no, not adult content, as in "made by VALVe in the age when those polygons would never be seen", like HL2 viewmodels), you can use roll as sway.

KM-CM commented:
Also, important note. So important I'm leaving a separate comment (jk I just only remembered now). `FrameTime()` IS FAULTY SOMETIMES! I remember that weapons behaved like 2x faster in multiplayer (on my base which used it), so be careful. The code I wrote prevents ANY possible issue with `FrameTime()`, by not using it.

KM-CM commented:
Also, I never really liked the base. And gave up on it since it conflicts with my addon. But I am still doing this because of the respect I have for it :)

buu342 commented:
I'm not really trying to make Far Cry 3: The SWEP Base, I'm trying to make _my_ base with features that I wanted. Some of those features happen to also be stuff that I enjoyed about Far Cry 3's weapons.

KM-CM committed:
EVEN more reliable systems! :)

In reality it's just a fix of 00e92675f95f4bec6d3984be7b5feee838b860d0
